OWID/OWID_Visualize_Population_of_Different_Age_Groups.ipynb

```diff
@@ -0,0 +1,405 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e2d4cef3",
+   "metadata": {},
+   "source": [
+    "**Tags:** #dash #dashboard #plotly #naas #asset #analytics #dropdown #callback #bootstrap #snippet"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1edaba32",
+   "metadata": {},
+   "source": [
+    "**Author:** [Zihui Ouyang](https://www.linkedin.com/in/zihui-ouyang-539626227/)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "371b301c",
+   "metadata": {},
+   "source": [
+    "**Description:** This notebook creates an interactive plot using Dash app infrastructure with OWID's popultion by age group data."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "faff9224",
+   "metadata": {},
+   "source": [
+    "**References:**\n",
+    "- https://ourworldindata.org/grapher/population-by-age-group-with-projections\n",
+    "- https://stackoverflow.com/questions/70886359/dash-python-making-subplots-when-multiple-parameters-are-selected\n",
+    "- https://dash-example-index.herokuapp.com/line-charts"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9f37978d",
+   "metadata": {},
+   "source": [
+    "## Input"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d634fde7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    import dash\n",
+    "    import os\n",
+    "except:\n",
+    "    !pip install dash --user\n",
+    "    import dash\n",
+    "try:\n",
+    "    import dash_bootstrap_components as dbc\n",
+    "except:\n",
+    "    !pip install dash_bootstrap_components --user\n",
+    "    import dash_bootstrap_components as dbc\n",
+    "import pandas as pd\n",
+    "from dash import Dash, html, dcc, callback, Output, Input\n",
+    "import plotly.express as px"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "665854f7",
+   "metadata": {},
+   "source": [
+    "### Setup Variables\n",
+    "- `DASH_PORT`: specify a port number for Dash\n",
+    "- `url`: URL to get data from Excel"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "929e8701",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DASH_PORT = 8050\n",
+    "url = \"https://population.un.org/wpp/Download/Files/1_Indicators%20(Standard)/EXCEL_FILES/2_Population/WPP2022_POP_F02_1_POPULATION_5-YEAR_AGE_GROUPS_BOTH_SEXES.xlsx\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0159d636",
+   "metadata": {},
+   "source": [
+    "## Model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a83dcac2",
+   "metadata": {},
+   "source": [
+    "### Initialize Dash App"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b5f389ed",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "app = dash.Dash(\n",
+    "    title = \"Population composition\",\n",
+    "    requests_pathname_prefix=f'/user/{os.environ.get(\"JUPYTERHUB_USER\")}/proxy/{DASH_PORT}/',\n",
```

Comment:
@makingsurgeon Would you mind using the Naas [secrets](https://site.naas.ai/docs/ecosystem/jobs/secret) feature here because `os.environ` is not supported in the naas platform.

Reply:
I made an attempt to fix this, not sure if it's correct, though.